lib/drivers/tandemDriver.js

```diff
@@ -1131,14 +1219,33 @@ module.exports = function (config) {
         if (result.payload.header_log_seq_no === end_seq) {
           debug('fetched all records');
           data.log_records = retval;
-          callback(null, data);
+
+          if(data.firmware_version >= COMMANDS.LOG_ENTRY_SEQ_MULTI_STOP_DUMP.version)
+            tandemCommand(COMMANDS.LOG_ENTRY_SEQ_MULTI_STOP_DUMP, null, function (err) {
+              if(err) {
+                callback(err,null);
+              }
+              else{
+                callback(null,data);
+              }
+            });
+          else {
+            callback(null, data);
+          }
         }
       }
     }
 
     end_seq = data.end_seq;
     start_seq = data.start_seq;
-    tandemLogRequester(start_seq, end_seq, progress, iterate);
+    debug('Firmware version is #',data.firmware_version);
+    if(data.firmware_version >= COMMANDS.LOG_ENTRY_SEQ_MULTI_REQ.version) {
+      // woohoo, we can use multi-record downloads!
+      multiLogRequester(start_seq, end_seq+1, progress, iterate);
```

Comment:
When we send a multi-record request we send the number of records to retrieve instead of just the index number. I've moved where the one gets added to a place where it makes more sense, and also added a comment to explain.